### ISSUE TITLE
Show which branches have unmerged changes

### ISSUE THREAD

**Issue by c2d7fa** [jesseduffield/lazygit#1147]
**Is your feature request related to a problem? Please describe.**

I use feature branches. Sometimes I work on one feature before switching to another. With lots of branches, I sometimes lose track of which branches still need work done, and which branches have already been merged and can be safely deleted.

**Describe the solution you'd like**

It would be nice if there was some indicator in the "Local Branches" pane for the branches that have not yet been merged into the branch that's currently checked out.

**Additional context**

I tried sketching out a patch [(WIP)](https://github.com/c2d7fa/lazygit), and came up with this:

![screenshot](https://user-images.githubusercontent.com/56928508/105547691-0e398a80-5cff-11eb-8ca4-a65eb9245bcb.png)

The deltas next to some branches indicate that the branch has not yet been merged into the checked out branch. Branches that don't have deltas have been merged and can be safely deleted.

I'd be happy to try and turn this into a proper pull request if this seems like a good fit for the project.

By the way, thanks for the great work! Lazygit is a really cool project!

**Comment by daviddwlee84** [comment]
An optional ahead/behind display relative to the base branch could help with this use case.

For a hypothetical example, suppose two feature branches both track their own remote branches and are fully synced with them. Both are 20 commits behind `main`, but one branch's history is already contained in `main`, while the other still has 3 commits that `main` does not contain. With the current behind-only display, both show `✓` and `↓20`, so scanning the branch list does not reveal that difference.

An opt-in display could make it visible:

```text
                         relative to main
feature-a  ✓             ↓20 ↑0
feature-b  ✓             ↓20 ↑3
```

Could `showDivergenceFromBaseBranch` offer a mode that shows both counts, while preserving the existing defaults? This would compare against the detected base branch, which is a slightly different scope from this issue's proposal to compare against the currently checked-out branch.

I found the earlier discussion in [#3536](https://github.com/jesseduffield/lazygit/issues/3536#issuecomment-2094065324), where the usefulness of ahead counts was acknowledged but the extra visual noise was a concern. Since [#4785](https://github.com/jesseduffield/lazygit/pull/4785) moved the base-divergence indicator to the right, perhaps an optional two-count mode would now fit better while keeping it separate from upstream status.

The counts would describe commit ancestry only. A nonzero ahead count can remain after a squash or rebase merge, so this should not be presented as a definitive PR-merged or safe-to-delete indicator. The benefit is being able to distinguish branches with unique commits from branches whose history is already contained in the base, directly in the list.

<details>
<summary>Possible configuration and implementation pointers</summary>

For example, users could opt in through their LazyGit `config.yml`:

```yaml
gui:
  # Proposed new value; NOT currently supported.
  showDivergenceFromBaseBranch: aheadAndBehind
```

`aheadAndBehind` is only a suggested name. The existing `none`, `onlyArrow`, and `arrowAndNumber` modes would keep their current behavior; users could switch back to `arrowAndNumber` for behind-only numbers.

Looking at the current source, these seem like the relevant places to extend:

- [`pkg/config/user_config.go`](https://github.com/jesseduffield/lazygit/blob/d0ede21e9c1e13de255b78f28102450d4523d748/pkg/config/user_config.go#L187): add the new supported value to the config definition/schema and update the generated configuration documentation.
- [`pkg/commands/git_commands/branch_loader.go`](https://github.com/jesseduffield/lazygit/blob/d0ede21e9c1e13de255b78f28102450d4523d748/pkg/commands/git_commands/branch_loader.go#L148): retain both counts for the same selected base. The Git >= 2.41 path already parses both via `%(ahead-behind:...)`, but `selectBehindForBranch` returns only behind; the legacy `rev-list --left-right --count` query also returns both. Carry the pair through refreshes instead of discarding ahead.
- [`pkg/commands/models/branch.go`](https://github.com/jesseduffield/lazygit/blob/d0ede21e9c1e13de255b78f28102450d4523d748/pkg/commands/models/branch.go#L39): store both base-relative counts, with a way to distinguish a valid result from unknown/not-yet-loaded data. Currently `BehindBaseBranch == 0` also means not yet determined, so it would be misleading to render an explicit `↑0` before a successful calculation.
- [`pkg/gui/presentation/branches.go`](https://github.com/jesseduffield/lazygit/blob/d0ede21e9c1e13de255b78f28102450d4523d748/pkg/gui/presentation/branches.go#L253): extend `divergenceStr` to render the pair in the new mode, including ahead-only branches where behind is zero, while retaining the existing right-aligned placement.

The existing branch-loader, branch-presentation, and `show_divergence_from_base_branch` integration tests would be natural places to cover ahead-only, behind-only, diverged, equal, and unknown states, plus unchanged behavior for the existing modes.

These are implementation pointers rather than a tested patch; the exact config name and data representation would be up to the maintainers.

</details>

**Comment by stefanhaller** [comment]
> An optional ahead/behind display relative to the base branch could help with this use case.

I would still find this too noisy, and I also still think that the ahead count is not very useful; for an unmerged branch it's just the length of the branch, and I don't feel you need to see this information all the time.

For deciding whether a branch has been merged, a number that you have to compare to 0 doesn't feel like the best way to show it. I'd rather have the presence or absence of an icon for that, so something like the original suggestion with the `Δ` icon would work for me; I'd turn the logic around though and show the icon for merged branches, not for unmerged ones. Unmerged branches are the normal case, I don't want to see something for them; for merged branches it's useful as an indication that you can delete them.

(One thing that's slightly weird is that you'd see the icon the moment you create a new branch, before you made the first commit, but maybe that's fine. If you end up not making any commits, then this shows you that you can delete the branch again in case you forgot. This does actually happen to me occasionally, so I like the idea.)

BTW: the description above says the icon shows for branches that have not yet been merged _into the checked out branch_. This doesn't feel right to me, it should be the base branch, not the checked out branch. (See #5635 for why it can be difficult to determine what the base branch is (that's still unsolved), but that's the same problem we already have for the behind count.)

Another BTW: I don't really miss this feature myself, because I don't merge branches locally. I merge them on Github (or coworkers do), and then we delete the remote branch there as well; and I have the `fetch.prune` git config set to true, so the next time I fetch, I see a big red "upstream gone" next to the branch and know I can delete it. And even if the branch is not deleted upstream, I still see the purple "merged" status for the PR. So I suppose this feature is really mostly needed by people who don't use PRs at all?

**Comment by daviddwlee84** [comment]
That makes sense—a merged-only icon would work well for my use case.

I mainly work on personal projects with coding agents working in parallel worktrees. Those branches are merged locally into `main`, often without being pushed or opening PRs, so neither “upstream gone” nor PR status is available. The goal is simply to spot which local branches have already been integrated and can be considered for cleanup.

Comparing against the detected/configured base branch, rather than the checked-out branch, also sounds right to me.
